app/models/concerns/indicateable.rb

```diff
@@ -65,8 +65,7 @@ def reading(indicator, options = {})
       raise ArgumentError, "Unknown indicator #{indicator.inspect}. Expecting one of them: #{Nomen::Indicator.all.sort.to_sentence}."
     end
     read_at = options[:at] || Time.zone.now
-    indicator_name = indicator.name
-    readings.where(indicator_name: indicator.name).where('read_at <= ?', read_at).order(read_at: :desc).first
+    readings.where("indicator_name = ? AND read_at <= ?", indicator.name, read_at).max_by(&:read_at)
```

Comment:
`max_by` adds performance issue, because it requires all data of query before sorting them. `.order(...).first` limits data to one row only directly.